src/pages/index.js

```diff
@@ -8,80 +8,172 @@ import { initialCards, cardData, options , cardSelector } from "../utils/constan
+const avatarValidator = new FormValidator(options, avatarForm)
+avatarValidator.enableValidation();
+
+const imagePreviewPopup = new PopupWithImage("#preview-modal");
 imagePreviewPopup.setEventListeners();
 
-function handleImageClick(data) {
-  imagePreviewPopup.open(data);
-};
+const confirmationPopup = new PopupWithConfirmation({popupSelector: options.deletePopup});
+confirmationPopup.setEventListeners();
+
+const avatarPopup = new PopupWithForm("#avatar-modal", handleAvatarChange);
+avatarPopup.setEventListeners();
 
+const api = new Api({
+  url: "https://around-api.en.tripleten-services.com/v1/",
+  headers: {
+    authorization: "0d8f734d-caf1-45e3-b9d3-764b4099955a",
+    "Content-Type": "application/json"
+  }
+}); 
+
+let cardSection;
+
+const renderCard = (data) => {
+  const card = createCard(data);
+  cardSection.addItem(card);
+};
 
-function createCard(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick)
-  return card.getView()
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cards]) => {
+    userInfo.setUserInfo(userData);
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: renderCard,
+      },
+      options.cardList
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+const userInfo = new UserInfo(
+  profileTitle, 
+  profileDescription,
+  profileAvatar
+  );
+
+function createCard(data) {
+  const cardElement = new Card(
+    {
+      data,
+      handleImageClick: (imgData) => {
+        imagePreviewPopup.open(imgData);
+      },
+      handleDeleteClick: () => {
+        confirmationPopup.open(() => {
+          confirmationPopup.renderLoading(true);
+          api.deleteCard(data._id)
+            .then(() => {
+              cardElement.removeCard();
+              confirmationPopup.close();
+            })
+            .catch((err) => console.log(`An error occured: ${err}`))
+            .finally(() => confirmationPopup.renderLoading(false));
+        });
+      },
+      handleLikeClick: () => {
+        if (cardElement.isLiked()) {
+          api
+            .removeLike(cardElement._cardId)
+            .then((res) => {
+              cardElement.setLikes(res.isLiked);
+            })
+            .catch((err) => {
+              console.log(`An error occured: ${err}`);
+            });
+        } else {
+          api
+            .addLike(cardElement._cardId)
+            .then((res) => {
+              cardElement.setLikes(res.isLiked);
+            })
+            .catch((err) => {
+              console.log(`An error occured: ${err}`);
+            });
+        }
+      },
+      userId: userInfo.getId(),
+    },
+    options.cardTemplate
+  );
+  return cardElement.getView();
 }
 
+
 function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data.name, data.profession)
-  editProfilePopup.close()
+  api.updateProfile(data.name, data.profession)
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```